src/main/steps/applicant1/confirm-your-joint-application/content.ts

```diff
@@ -1,158 +1,176 @@
+import config from 'config';
+
 import { getFormattedDate } from '../../../app/case/answers/formatDate';
-import { getAnswerRows } from '../../../app/case/answers/getAnswerRows';
-import { Checkbox } from '../../../app/case/case';
-import { ChangedNameHow, YesOrNo } from '../../../app/case/definition';
+import { CaseWithId, Checkbox } from '../../../app/case/case';
+import { YesOrNo } from '../../../app/case/definition';
 import { TranslationFn } from '../../../app/controller/GetController';
 import { FormContent } from '../../../app/form/Form';
 import { isFieldFilledIn } from '../../../app/form/validation';
-import { connectionBulletPointsTextForSoleAndJoint } from '../../../app/jurisdiction/bulletedPointsContent';
+import { connectionBulletPointsTextForRespondent } from '../../../app/jurisdiction/bulletedPointsContent';
 import { jurisdictionMoreDetailsContent } from '../../../steps/applicant1/connection-summary/content';
-import { Sections } from '../../applicant1Sequence';
-import { moreDetailsComponent } from '../../applicant2/check-your-joint-application/content';
 import { CommonContent } from '../../common/common.content';
-import * as urls from '../../urls';
 
-const en = ({ isDivorce, partner, userCase, marriage, civilPartnership }: CommonContent) => ({
+const isSubmit = (isApplicant2: boolean, userCase: Partial<CaseWithId> | undefined): boolean => {
```

Comment:
how comes userCase can be undefined here?

Reply:
Actually, can userCase ever be undefined?  This has been confusing me for a while.  What situation can a userCase be undefined?

Reply:
no, as far as I'm aware it won't ever be undefined

Reply:
So why do we have the ? everywhere next to it?   OK so I will remove the undefined bit but it's even more confusing

Reply:
I think @velimd is addressing that

Reply:
I've changed it but it looks so weird now.  In a couple of places it looks like it cannot be undefined (userCase!)  and in others it looks like it can userCase?  Perhaps we should leave it as it it with possibly undefined, then create a ticket that removes it.  What do you think?

Reply:
I think Veli's working on getting rid of the ? so I would assume that would also fix this problem ^

Reply:
OK so is it better to leave it as it is showing it can be undefined - even though we know it never will be - or add the ! in certain places?  I prefer to leave as it is and let Veli remove all the undefined as opposed to mixing ? and ! in the code next to userCase

Reply:
its considered undefined currently because its an optional property (which im making it conpulsory), hence all the ?.